lib/MySQL_Thread.cpp

```diff
@@ -4091,6 +4091,128 @@ void MySQL_Thread::unregister_session(int idx) {
 }
 
 
+// this function was inline in MySQL_Thread::run()
+void MySQL_Thread::run___get_multiple_idle_connections(int& num_idles) {
+	int rc;
+	int i;
+	num_idles=MyHGM->get_multiple_idle_connections(-1, curtime-mysql_thread___ping_interval_server_msec*1000, my_idle_conns, SESSIONS_FOR_CONNECTIONS_HANDLER);
+	for (i=0; i<num_idles; i++) {
+		MySQL_Data_Stream *myds;
+		MySQL_Connection *mc=my_idle_conns[i];
+		MySQL_Session *sess=new MySQL_Session();
+		sess->mybe=sess->find_or_create_backend(mc->parent->myhgc->hid);
+
+		myds=sess->mybe->server_myds;
+		myds->attach_connection(mc);
+		myds->assign_fd_from_mysql_conn();
+		myds->myds_type=MYDS_BACKEND;
+
+		sess->to_process=1;
+		myds->wait_until=curtime+mysql_thread___ping_timeout_server*1000;	// max_timeout
+		mc->last_time_used=curtime;
+		myds->myprot.init(&myds, myds->myconn->userinfo, NULL);
+		sess->status=PINGING_SERVER;
+		myds->DSS=STATE_MARIADB_PING;
+		register_session_connection_handler(sess,true);
+		int rc=sess->handler();
+		if (rc==-1) {
+			unsigned int sess_idx=mysql_sessions->len-1;
+			unregister_session(sess_idx);
+			delete sess;
+		}
+	}
+	processing_idles=true;
+	last_processing_idles=curtime;
+}
+
+// this function was inline in MySQL_Thread::run()
+void MySQL_Thread::ProcessAllMyDS_BeforePoll() {
+	unsigned int n;
+	for (n = 0; n < mypolls.len; n++) {
```

Comment:
In the other function `n` was declared inline within the `for`, just to keep consistency.